File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int start=0;
        int i=0;
        while(i<nums.size()){
            if(nums[start]!=nums[i]){
                swap(nums[i++],nums[++start]);
            }
            else
            i++;
        }
        return start+1;
        // int k = max(0,start-1);
        // return k;
    }
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
     int count = 1;
     for(int i=0;i<nums.length-1;){
        int k = i+1;
        while(k<nums.length){
            if(nums[k]==nums[i]){
                k++;
            }
            else{
                nums[count] = nums[k];
                count+=1;
                i=k;
            }
        }
        if(i!=k)
            i+=1;
     }
     return count;   
    }
}